### PR TITLE
fix: Better method to detect helm chart version in template tests

### PR DESCRIPTION
**Pull request by harshshahsumo**
This change finds a new method using regex for updating the helm chart version without affecting any other version that matches it
Find the [Ticket](https://sumologic.atlassian.net/browse/OSC-665)

### Checklist
- [X] Changelog updated or skip changelog label added
- [ ] Documentation updated
- [ ] Template tests added for new features
- [ ] Integration tests added or modified for major features

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
Please add unit tests for this to prevent regression in case of change

File: tests/helm/goldenfile_test.go [hunk 1]
```diff
@@ -132,11 +133,172 @@ func runGoldenFileTest(t *testing.T, valuesFileName string, outputFileName strin
...
 func fixupRenderedYaml(yaml string, chartVersion string) string {
 	checksumRegex := regexp.MustCompile("checksum/config: [a-z0-9]{64}")
+	patternString := fmt.Sprintf(
+		`(?m)^(\s*app\.kubernetes\.io\/version:\s*"%s"\s*)$`+
+			`|^(\s*chart:\s*"sumologic-%s"\s*)$`+
+			`|^(\s*chart:\s*sumologic-%s\s*)$`+
+			`|^(\s*client:\s*k8s_%s\s*)$`+
+			`|^(\s*value:\s*"%s"\s*)$`,
+		chartVersion, chartVersion, chartVersion, chartVersion, chartVersion)
+	pattern := regexp.MustCompile(patternString)
```
Comment (sumo-drosiek) [inline tests/helm/goldenfile_test.go:143]: As it's correct, it's not so easy to understand the regex. I would recommend to create lis of regexes, and then iterate over them and replace in the output, or even just replace bare strings. I would want to maintain it in this form after all

File: tests/helm/goldenfile_test.go [hunk 2]
```diff
@@ -132,11 +133,167 @@ func runGoldenFileTest(t *testing.T, valuesFileName string, outputFileName strin
 // expected templates
 func fixupRenderedYaml(yaml string, chartVersion string) string {
 	checksumRegex := regexp.MustCompile("checksum/config: [a-z0-9]{64}")
+	replacements := map[string]string{
+		fmt.Sprintf("app.kubernetes.io/version: \"%s\"", chartVersion): "app.kubernetes.io/version: \"%CURRENT_CHART_VERSION%\"",
+		fmt.Sprintf("chart: \"sumologic-%s\"", chartVersion):           "chart: \"sumologic-%CURRENT_CHART_VERSION%\"",
+		fmt.Sprintf("chart: sumologic-%s", chartVersion):               "chart: sumologic-%CURRENT_CHART_VERSION%",
+		fmt.Sprintf("client: k8s_%s", chartVersion):                    "client: k8s_%CURRENT_CHART_VERSION%",
+		fmt.Sprintf("value: \"%s\"", chartVersion):                     "value: \"%CURRENT_CHART_VERSION%\"",
+	}
```
Comment (sumo-drosiek) [inline tests/helm/goldenfile_test.go:142]:
```suggestion
	replacements := []string{
		"app.kubernetes.io/version: \"%s\"",
		"chart: \"sumologic-%s\"", chartVersion):           "chart: \"sumologic-%CURRENT_CHART_VERSION%\"",
		fmt.Sprintf("chart: sumologic-%s", chartVersion):               "chart: sumologic-%CURRENT_CHART_VERSION%",
		fmt.Sprintf("client: k8s_%s", chartVersion):                    "client: k8s_%CURRENT_CHART_VERSION%",
		fmt.Sprintf("value: \"%s\"", chartVersion):                     "value: \"%CURRENT_CHART_VERSION%\"",
	}
```

Comment (sumo-drosiek) [inline tests/helm/goldenfile_test.go:142]:
```suggestion
	replacements :=[]string{
		"app.kubernetes.io/version: \"%s\"",
		"chart: \"sumologic-%s\"",
		"chart: sumologic-%s",
		"client: k8s_%s",
		"value: \"%s\"",
	}
```

File: tests/helm/goldenfile_test.go [hunk 3]
```diff
@@ -132,11 +133,167 @@ func runGoldenFileTest(t *testing.T, valuesFileName string, outputFileName strin
...
 
 	output := yaml
 	output = strings.ReplaceAll(output, releaseName, "RELEASE-NAME")
-	output = strings.ReplaceAll(output, chartVersion, "%CURRENT_CHART_VERSION%")
+	for oldString, newString := range replacements {
+		output = strings.ReplaceAll(output, oldString, newString)
+	}
```
Comment (sumo-drosiek) [inline tests/helm/goldenfile_test.go:148]:
```suggestion
	for_, r := range replacements {
		output = strings.ReplaceAll(output, fmt.Sprintf(r, chartVersion), fmt.Sprintf(r, "%CURRENT_CHART_VERSION%"))
	}
```

**Review by sumo-drosiek — COMMENTED** [review]
I believe this is better. It's following more DRY concept

File: tests/helm/goldenfile_test.go (hunk 1 above)
Comment (harshshahsumo) [reply tests/helm/goldenfile_test.go]: addressed

File: tests/helm/goldenfile_test.go (hunk 2 above)
Comment (harshshahsumo) [reply tests/helm/goldenfile_test.go]: addressed

File: tests/helm/goldenfile_test.go (hunk 3 above)
Comment (harshshahsumo) [reply tests/helm/goldenfile_test.go]: addressed

File: tests/helm/goldenfile_test.go [hunk 4]
```diff
@@ -132,11 +133,173 @@ func runGoldenFileTest(t *testing.T, valuesFileName string, outputFileName strin
...
 func fixupRenderedYaml(yaml string, chartVersion string) string {
 	checksumRegex := regexp.MustCompile("checksum/config: [a-z0-9]{64}")
-
+	// replacements := map[string]string{
+	// 	fmt.Sprintf("app.kubernetes.io/version: \"%s\"", chartVersion): "app.kubernetes.io/version: \"%CURRENT_CHART_VERSION%\"",
+	// 	fmt.Sprintf("chart: \"sumologic-%s\"", chartVersion):           "chart: \"sumologic-%CURRENT_CHART_VERSION%\"",
+	// 	fmt.Sprintf("chart: sumologic-%s", chartVersion):               "chart: sumologic-%CURRENT_CHART_VERSION%",
+	// 	fmt.Sprintf("client: k8s_%s", chartVersion):                    "client: k8s_%CURRENT_CHART_VERSION%",
+	// 	fmt.Sprintf("value: \"%s\"", chartVersion):                     "value: \"%CURRENT_CHART_VERSION%\"",
+	// }
```
Comment (echlebek) [inline tests/helm/goldenfile_test.go:142]: Commented out code should be removed

File: tests/helm/goldenfile_test.go [hunk 5]
```diff
@@ -132,11 +133,173 @@ func runGoldenFileTest(t *testing.T, valuesFileName string, outputFileName strin
...
+	replacements := []string{
+		"app.kubernetes.io/version: \"%s\"",
+		"chart: \"sumologic-%s\"",
+		"chart: sumologic-%s",
+		"client: k8s_%s",
+		"value: \"%s\"",
```
Comment (echlebek) [inline tests/helm/goldenfile_test.go:148]: Use %q instead of %s for items that need to be surrounded in quotes, as this allows you to avoid escapes here.

**Review by echlebek — CHANGES_REQUESTED** [review]
Almost good to go, a few things need fixing still

File: tests/helm/goldenfile_test.go (hunk 4 above)
Comment (harshshahsumo) [reply tests/helm/goldenfile_test.go]: done!

File: tests/helm/goldenfile_test.go (hunk 5 above)
Comment (harshshahsumo) [reply tests/helm/goldenfile_test.go]: addressed

**Review by echlebek — APPROVED** [review]
Looks good :ship:

**Pull request merged**